sycl/include/sycl/types.hpp

```diff
@@ -1024,42 +1057,39 @@ template <typename Type, int NumElements> class vec {
-        Ret.setValue(I, (DataT)(vec_data<DataT>::get(getValue(                 \
+        Ret.setValue(I, (DataT)(vec_data<DataT>::get(Lhs.getValue(             \
                             I)) BINOP vec_data<DataT>::get(Rhs.getValue(I)))); \
     return Ret;                                                                \
   }                                                                            \
-  template <typename T>                                                        \
-  typename std::enable_if_t<                                                   \
-      std::is_convertible_v<DataT, T> &&                                       \
-          (std::is_fundamental_v<vec_data_t<T>> ||                             \
-           detail::is_half_or_bf16_v<typename std::remove_const_t<T>>),        \
-      vec>                                                                     \
-  operator BINOP(const T & Rhs) const {                                        \
-    return *this BINOP vec(static_cast<const DataT &>(Rhs));                   \
+  friend vec operator BINOP(const vec &Lhs, const DataT &Rhs) {                \
+    return Lhs BINOP vec(Rhs);                                                 \
   }                                                                            \
-  vec &operator OPASSIGN(const vec & Rhs) {                                    \
-    *this = *this BINOP Rhs;                                                   \
-    return *this;                                                              \
+  friend vec operator BINOP(const DataT &Lhs, const vec &Rhs) {                \
+    return vec(Lhs) BINOP Rhs;                                                 \
+  }                                                                            \
+  friend vec &operator OPASSIGN(vec & Lhs, const vec & Rhs) {                  \
+    Lhs = Lhs BINOP Rhs;                                                       \
+    return Lhs;                                                                \
   }                                                                            \
   template <int Num = NumElements>                                             \
-  typename std::enable_if_t<Num != 1, vec &> operator OPASSIGN(                \
-      const DataT & Rhs) {                                                     \
-    *this = *this BINOP vec(Rhs);                                              \
-    return *this;                                                              \
+  friend typename std::enable_if_t<Num != 1, vec &> operator OPASSIGN(         \
+      vec & Lhs, const DataT & Rhs) {                                          \
+    Lhs = Lhs BINOP vec(Rhs);                                                  \
+    return Lhs;                                                                \
   }
-#endif // defined(__INTEL_PREVIEW_BREAKING_CHANGES)
-
-#if !defined(__INTEL_PREVIEW_BREAKING_CHANGES)
```

Comment:
we are trying to avoid using `#else` for long blocks.    It makes the closing comment easier to understand.

For example, you've changed this so now it ends like so: 

`#endif // defined(__INTEL_PREVIEW_BREAKING_CHANGES)`

Which would lead the casual reader to think ('oh, this block above is for the preview") which is wrong. The block immediately above is the ELSE clause of the preview.   

I know it seems pedantic, but especially with this preview stuff where we have long blocks of semi-repeating code, let's try to avoid `#else`

Reply:
Personally, I disagree with that, but it is not a hill I want to die on. We could simply change the comment on the `#endif` to make it clearer and the `#else`, I would argue, makes the association clearer between the two blocks. Either way, I've split them up.